java-checks/src/main/java/org/sonar/java/checks/DeadStoreCheck.java

```diff
@@ -176,12 +178,38 @@ private static void handleIdentifier(Set<Symbol> out, Set<Tree> assignmentLHS, I
 
   private void handleVariable(Set<Symbol> out, VariableTree localVar) {
     Symbol symbol = localVar.symbol();
-    if (localVar.initializer() != null && !out.contains(symbol)) {
-      createIssue(((VariableTreeImpl) localVar).equalToken(), localVar.initializer(), symbol);
+    ExpressionTree initializer = localVar.initializer();
+    if (initializer != null && !isUsualDefaultValue(initializer) && !out.contains(symbol)) {
+      createIssue(((VariableTreeImpl) localVar).equalToken(), initializer, symbol);
     }
     out.remove(symbol);
   }
 
+  private static boolean isUsualDefaultValue(ExpressionTree initializer) {
+    ExpressionTree expr = ExpressionUtils.skipParentheses(initializer);
```

Comment:
Wondering if using a switch on kind wouldn't help to make things a bit clearer here.